aiodocker/docker.py

```diff
@@ -113,11 +113,24 @@ def __init__(self,
         data = await self._query_json("version")
         return data
 
-    async def pull(self, image, *, stream=False):
+    # maybe discard future
+    async def pull(self, image, auth=None, stream=False):
+
+        headers = {"content-type": "application/json"}
+        if auth:
+            if isinstance(auth, dict) and 'auth' in auth:
+                registry, p, _ = image.partition('/')
```

Comment:
use auth with private register,  image name must be include register name!  here to check it ...
it is better to use docker.images.pull to install this interface ,,  I think it maybe delete in future ...